MediaCapabilities should respect allowed containers and codecs in Captive Portal mode

jernoble opened the pull request:
#### aa2c0e352a3c9b9a0c918dec9b64b9afd6fa553a
<pre>
MediaCapabilities should respect allowed containers and codecs in Captive Portal mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=242145">https://bugs.webkit.org/show_bug.cgi?id=242145</a>
&lt;rdar://92419877&gt;

Reviewed by Eric Carlson.

Pipe the allowedMediaContainerTypes and allowedMediaCodecTypes document Settings into
MediaConfiguration, and from there MediaEngineSupportParameters, when determining
decoding capabilities.

Previously, these settings were enforced in MediaSource itself, but for MediaCapabilities,
they must also be checked by MediaPlayerPrivateMediaSourceAVFObjC.

(Slightly changing commit message in order to force-push/retry building.)

* LayoutTests/media/mediacapabilities/mediacapabilities-allowed-codecs-expected.txt: Added.
* LayoutTests/media/mediacapabilities/mediacapabilities-allowed-codecs.html: Added.
* LayoutTests/media/mediacapabilities/mediacapabilities-allowed-containers-expected.txt: Added.
* LayoutTests/media/mediacapabilities/mediacapabilities-allowed-containers.html: Added.
* Source/WebCore/Modules/mediacapabilities/MediaCapabilities.cpp:
(WebCore::gatherDecodingInfo):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::supportsTypeAndCodecs):
* Source/WebCore/platform/graphics/cocoa/MediaEngineConfigurationFactoryCocoa.cpp:
(WebCore::computeMediaCapabilitiesInfo):
* Source/WebCore/platform/mediacapabilities/MediaConfiguration.h:
(WebCore::MediaConfiguration::isolatedCopy const):
(WebCore::MediaConfiguration::isolatedCopy):
(WebCore::MediaConfiguration::encode const):
(WebCore::MediaConfiguration::decode):

Canonical link: <a href="https://commits.webkit.org/252233@main">https://commits.webkit.org/252233@main</a>
</pre>

eric-carlson reviewed (approved):
I love a PR with more test than code change!

jernoble committed:
MediaCapabilities should respect allowed containers and codecs in Captive Portal mode
https://bugs.webkit.org/show_bug.cgi?id=242145
<rdar://92419877>

Reviewed by Eric Carlson.

Pipe the allowedMediaContainerTypes and allowedMediaCodecTypes document Settings into
MediaConfiguration, and from there MediaEngineSupportParameters, when determining
decoding capabilities.

Previously, these settings were enforced in MediaSource itself, but for MediaCapabilities,
they must also be checked by MediaPlayerPrivateMediaSourceAVFObjC.

(Slightly changing commit message in order to force-push/retry building.)

* LayoutTests/media/mediacapabilities/mediacapabilities-allowed-codecs-expected.txt: Added.
* LayoutTests/media/mediacapabilities/mediacapabilities-allowed-codecs.html: Added.
* LayoutTests/media/mediacapabilities/mediacapabilities-allowed-containers-expected.txt: Added.
* LayoutTests/media/mediacapabilities/mediacapabilities-allowed-containers.html: Added.
* Source/WebCore/Modules/mediacapabilities/MediaCapabilities.cpp:
(WebCore::gatherDecodingInfo):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::supportsTypeAndCodecs):
* Source/WebCore/platform/graphics/cocoa/MediaEngineConfigurationFactoryCocoa.cpp:
(WebCore::computeMediaCapabilitiesInfo):
* Source/WebCore/platform/mediacapabilities/MediaConfiguration.h:
(WebCore::MediaConfiguration::isolatedCopy const):
(WebCore::MediaConfiguration::isolatedCopy):
(WebCore::MediaConfiguration::encode const):
(WebCore::MediaConfiguration::decode):

Canonical link: https://commits.webkit.org/252233@main
